src/hyperiond-starter/configHandle.cpp

```diff
@@ -0,0 +1,291 @@
+		quint16 protoPort = obj["protoServer"].toObject().take("port").toInt();
+		quint16 webPort = obj["webConfig"].toObject().take("port").toInt();
+
+		_usedPorts << jsonPort << protoPort << webPort;
+	}
+}
+
+void configHandle::processConfigChange(QString configPath)
+{
+	QString config = readFile(configPath);
+	qDebug() << "processConfigChange" << configPath;
+	QJsonDocument doc = QJsonDocument::fromJson(config.toUtf8());
+	QJsonArray arr = doc.object()["instances"].toArray();
+
+	for(int i = 0; i<arr.size(); ++i)
+	{
+		QString pName = arr[i].toObject().take("instName").toString();
+		QString cName = convertToCName(pName);
+
+		if(!configExists(cName))
+		{
+			if(createConfig(cName,pName))
+			{
+				_process.createProcess(cName);
+				if(arr[i].toObject().take("enabled").toBool())
+					_process.startProcessByCName(cName);
+			}
+		}
+		else
+		{
+			_process.createProcess(cName);
+			if(arr[i].toObject().take("enable").toBool())
+			{
+				_process.startProcessByCName(cName);
+			}
+			else
+			{
+				_process.stopProcessByCName(cName);
+			}
+		}
+	}
+}
+
+QString configHandle::convertToCName(QString pName)
+{
+	pName = pName.trimmed();
+	while(pName.startsWith("."))
+		pName.remove(0,1);
+
+	if(!pName.endsWith(".json"))
+		pName.append(".json");
+
+	return pName;
+}
+
+void configHandle::fileChanged(const QString path)
+{
+	processConfigChange(path);
+}
+
+//Snippet from : https://forum.qt.io/post/393109
+void configHandle::modifyJsonValue(QJsonValue& destValue, const QString& path, const QJsonValue& newValue)
+{
+    const int indexOfDot = path.indexOf('.');
+    const QString dotPropertyName = path.left(indexOfDot);
+    const QString dotSubPath = indexOfDot > 0 ? path.mid(indexOfDot + 1) : QString();
+
+    const int indexOfSquareBracketOpen = path.indexOf('[');
+    const int indexOfSquareBracketClose = path.indexOf(']');
+
+    const int arrayIndex = path.mid(indexOfSquareBracketOpen + 1, indexOfSquareBracketClose - indexOfSquareBracketOpen - 1).toInt();
+
+    const QString squareBracketPropertyName = path.left(indexOfSquareBracketOpen);
+    const QString squareBracketSubPath = indexOfSquareBracketClose > 0 ? (path.mid(indexOfSquareBracketClose + 1)[0] == '.' ? path.mid(indexOfSquareBracketClose + 2) : path.mid(indexOfSquareBracketClose + 1)) : QString();
+
+    // determine what is first in path. dot or bracket
+    bool useDot = true;
+    if (indexOfDot >= 0) // there is a dot in path
+    {
+        if (indexOfSquareBracketOpen >= 0) // there is squarebracket in path
+        {
+            if (indexOfDot > indexOfSquareBracketOpen)
+                useDot = false;
+            else
+                useDot = true;
+        }
+        else
+            useDot = true;
+    }
+    else
+    {
+        if (indexOfSquareBracketOpen >= 0)
+            useDot = false;
+        else
+            useDot = true; // acutally, id doesn't matter, both dot and square bracket don't exist
+    }
+
```

Comment:
you can replace the while  block 145 to 164 with this:
```
//            there is a dot and squarebracket in path, then determine whats first, otherwise use dot when no bracket
bool useDot = (indexOfDot >= 0 && indexOfSquareBracketOpen >= 0) ? (indexOfDot <= indexOfSquareBracketOpen) : (indexOfSquareBracketOpen < 0);
```